Improved Note-On performance

KKQ-KKQ opened the pull request:
I found that calculating a hash of the ModKey is the bottleneck.
I added a variable to cache ModKey's hash.

KKQ-KKQ commented:
I found ModKey hash function doesn't process its region.
I added the process.

paulfd commented on src/sfizz/utility/StringViewHelpers.h:
```diff
@@ -101,7 +101,12 @@ uint64_t hashNumber(Int i, uint64_t h = Fnv1aBasis)
     static_assert(std::is_arithmetic<Int>::value,
                   "The hashed object must be of arithmetic type");
-    return hash(absl::string_view(reinterpret_cast<const char*>(&i), sizeof(i)), h);
+    union {
```
Why was this needed? Some alignment issue?

KKQ-KKQ replied:
This block is not needed but this results in the better performance.
I changed this block because I found hashing process is costing a lot at first. but finally I found it is not the core issue.
The core issue is that the hash of ModKey region is not being calculated.

paulfd replied:
I see thanks

paulfd commented:
Thanks a lot for the PR and for catching this! I will take some time to review it better hopefully this WE. I had another issue a long time ago where I also needed to cache the key/id hashes for lookups in the modulation matrix...

paulfd commented:
Would it be okay that I reformat and squash the commits a bit to match the usual code style? I'll keep your authorship.

KKQ-KKQ commented:
No problem.
I am very new to co-working and I’m still getting used to using git.

KKQ-KKQ committed:
Improved note-on performance

- Compute the hash value on ModKey construction and memorize it
- Fix missing region ID in the hash value of ModKey

paulfd commented:
No worries, thanks a lot! I mostly put all commits into a single one (which is called "squashing") to make it easier to follow the history.
